components/tidb_query_datatype/src/codec/mysql/json/json_contains.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+
+use super::{super::Result, JsonRef, JsonType};
+
+impl<'a> JsonRef<'a> {
+    /// `json_contains` is the implementation for JSON_CONTAINS in mysql
+    /// <https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-contains>
+    /// See `ContainsBinary()` in TiDB `json/binary_function.go`
```

Comment:
Sorry, I could not found the function `ContainsBinary()` in TiDB `json/binary_function.go` .

Reply:
I change binary_function.go to types/json_binary_functions.go